@here/olp-sdk-core/lib/utils/MultiPartUploadWrapper.ts

```diff
@@ -177,9 +221,211 @@ export class MultiPartUploadWrapper {
+                            contentLength: buffer.byteLength,
+                            contentType: opts.contentType,
+                            partNumber: chunkNumber,
+                            url: startMultipartResponse.uploadPartUrl
+                        })
+                        .then(uploadPartResponse => {
+                            parts.push({
+                                id: uploadPartResponse.id,
+                                number: chunkNumber
+                            });
+
+                            uploadedChunks++;
+
+                            opts.onStatus &&
+                                opts.onStatus({
+                                    chunkId: uploadPartResponse.id,
+                                    chunkNumber,
+                                    chunkSize: buffer.byteLength,
+                                    totalChunks,
+                                    uploadedChunks
+                                });
+                        })
+                );
+            }
+
+            await Promise.all(promises);
+        }
+
+        await request.completeMultipartUpload({
+            parts,
+            url: startMultipartResponse.completeUrl,
+            multipartToken: startMultipartResponse.multipartToken,
+            layerId: opts.layerId,
+            billingTag: opts.billingTag
+        });
+
+        blobData.finally && blobData.finally();
+
         return STATUS_CODES.NO_CONTENT;
     }
+
+    private isStartMultipartResponseValid(obj: {
+        multipartToken?: string;
+        uploadPartUrl?: string;
+        completeUrl?: string;
+        statusUrl?: string;
+    }): boolean {
+        return (
+            obj.multipartToken !== undefined ||
+            (obj.uploadPartUrl !== undefined && obj.completeUrl !== undefined)
+        );
+    }
+
+    private isBlobData(input: any): boolean {
+        return (
+            typeof input.size === "function" &&
+            typeof input.readBytes === "function"
+        );
+    }
+
+    private async prepareBlobData(
+        input: string | File | Blob | ArrayBufferLike | BlobData
+    ): Promise<BlobData> {
+        if (this.isBlobData(input)) {
+            return input as BlobData;
+        }
+
+        if (typeof Buffer !== "undefined") {
+            // node.js
+            if (Buffer.isBuffer(input)) {
+                return new BufferData(input);
+            }
+
+            if (typeof input === "string") {
+                return NodeFileData.fromPath(input);
+            }
+        }
+
+        // browser
+        if (typeof input === "string") {
+            throw new Error("Unsupported input");
+        }
+
+        return input instanceof Blob
+            ? new WebData(input)
+            : new BufferData(input as ArrayBuffer);
+    }
+
+    private validateChunkSize(size = 0): number {
+        const minChunkSizeMiB = 5;
+        const mmxChunkSizeMiB = 5120;
```

Comment:
```suggestion
        const mmxChunkSizeMiB = 5 * 1024;
```
Should be easier to understand this way, what you actually mean.